file/provider/util.go

```diff
@@ -249,6 +249,51 @@ func FromProviderToLibVolume(vpcVolume *models.Share, logger *zap.Logger) (libVo
 		libVolume.Az = vpcVolume.Zone.Name
 	}
 	libVolume.CRN = vpcVolume.CRN
+
+	var respAccessPointlist = []provider.VolumeAccessPoint{}
+
+	shareTargetlist := vpcVolume.ShareTargets
+	if shareTargetlist != nil && len(*shareTargetlist) > 0 {
```

Comment:
Can we have separate if condition for these checks and for loop outside of look

Reply:
There is no else part for this logic so we have to populate the list if only we have share targets. Let me know if you some other flow in mind